### ISSUE TITLE
Error when setting a new image tag using Kustomize

### ISSUE THREAD

**Issue by ArjonBu** [argoproj/argo-cd#5137]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

I have a Kustomize app on ArgoCD called "staging" which includes a lot of k8s bases.
I have defined a new Kustomize version in ArgoCD. 

Project manifest:

```
project: default
source:
  repoURL: 'https://gitrepo/project/k8s.git'
  path: overlays/staging
  targetRevision: HEAD
  kustomize:
    version: v3.8.7
destination:
  server: 'https://kubernetes.default.svc'
syncPolicy: {}
```

I want to change an image tag using argocd command line. 

If I run: `argocd app set staging --kustomize-image grafana/grafana:7.3.4` it will return an error about a Kustomize patch which is not supported in the default Kustomize version (hence why I added a custom Kustomize version).

If I run:  `argocd app set staging --kustomize-image grafana/grafana:7.3.5 --kustomize-version v3.8.7` it will use my custom Kustomize version and this will work. 

The issue is `argocd app set staging --kustomize-image` command ignores my predefined Kustomize version in the project manifest and this will make argocd command line tool fail.

**Version**

```shell
argocd: v1.7.10+bcb05b0.dirty
  BuildDate: 2020-11-20T23:40:45Z
  GitCommit: bcb05b0c2e0f8006aa2d2abaf780e73c9e73c945
  GitTreeState: dirty
  GoVersion: go1.15.5
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v1.8.1+c2547dc
  BuildDate: 2020-12-10T02:59:21Z
  GitCommit: c2547dca95437fdbb4d1e984b0592e6b9110d37f
  GitTreeState: clean
  GoVersion: go1.14.12
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: v3.8.1 2020-07-16T00:58:46Z
  Helm Version: v3.4.1+gc4e7485
  Kubectl Version: v1.17.8
```

**Comment by jessesuen** [comment]
If I understand the bug, `argocd app set staging --kustomize-image grafana/grafana:7.3.4` clobbers the previously set `source.kustomize.version` value? If so this is a bug.

**Comment by ArjonBu** [comment]
Yes, that's the case. Unless I specify the kustomize version when running 
the argocd command line tool, the already defined kustomize version will be 
ignored.

**Comment by kshamajain99** [comment]
@ArjonBu Can you please share the error message that you are getting?

**Comment by ArjonBu** [comment]
The error message is one that I would get if i used the default kustomize 
version. I don't remember it right now but its a new patch plugin 
introduced in a later kustomize version.

**Comment by kshamajain99** [comment]
This bug has been fixed for >=v1.8.0 argocd cli by [this PR](https://github.com/argoproj/argo-cd/pull/4613/files#diff-95f228706705fc556fbb63b303d638ad4d4ddba8118a50dc83e344ac9b692d49R663). 
Will be closing this issue.
